akka-docs/src/main/paradox/typed/reliable-delivery.md

```diff
@@ -0,0 +1,358 @@
+and some of them may have already been processed but the fact that they were confirmed had not been stored
+yet. Meaning at-least once delivery.
+
+If a worker crashes or is stopped gracefully the unconfirmed messages will be redelivered to other workers.
+In that case some of these may already have been processed by the previous worker. Meaning at-least once delivery.
+
+## Sharding
+
+Reliable delivery between a producer actor sending messages to @ref:[sharded](cluster-sharding.md) consumer
+actor receiving the messages.
+
+![delivery-work-sharding-1.png](./images/delivery-sharding-1.png)
+
+and sending to another entity
+
+![delivery-work-sharding-2.png](./images/delivery-sharding-2.png)
+
+and sending from another producer (different node)
+
+![delivery-work-sharding-3.png](./images/delivery-sharding-3.png)
+
+The @apidoc[ShardingProducerController] should be used together with @apidoc[ShardingConsumerController].
+
+A producer can send messages via a `ShardingProducerController` to any `ShardingConsumerController`
+identified by an `entityId`. A single `ShardingProducerController` per `ActorSystem` (node) can be
+shared for sending to all entities of a certain entity type. No explicit registration is needed
+between the `ShardingConsumerController` and `ShardingProducerController`.
+
+The producer actor will start the flow by sending a `ShardingProducerController.Start`
+message to the `ShardingProducerController`.
+
+The `ShardingProducerController` sends `RequestNext` to the producer, which is then allowed
+to send one message to the `ShardingProducerController`. Thereafter the producer will receive a
+new `RequestNext` when it's allowed to send one more message.
+
+In the @apidoc[ShardingProducerController.RequestNext] message there is information about which entities
+that have demand. It is allowed to send to a new `entityId` that is not included in the `RequestNext.entitiesWithDemand`.
+If sending to an entity that doesn't have demand the message will be buffered. This support for buffering
+means that it is even allowed to send several messages in response to one `RequestNext` but it's recommended to
+only send one message and wait for next `RequestNext` before sending more messages.
+
+The producer and `ShardingProducerController` actors are supposed to be local so that these messages are
+fast and not lost.
+
+Similarly, on the consumer side the destination consumer actor will start the flow by sending an
+initial `ConsumerController.Start` message to the `ConsumerController`.
+
+There will be one `ShardingConsumerController` for each entity. Many unconfirmed messages can be in
+flight between the `ShardingProducerController` and each `ShardingConsumerController`, but it is 
+limited by a flow control window. The flow control is driven by the consumer side, which means that
+the `ShardingProducerController` will not send faster than the demand requested by the consumers.
+
+Example of `TodoList` entity (consumer):
```

Comment:
Example subheader above would separate it better.

A short introduction of the sample would be nice to set up for faster understanding the async juggle before diving into the sample code, something like: "The sharded entity is a todo list which uses an async database call to store its entire state on each change, and first when that completes replies to reliable delivery that the message was consumed"